neet/boolean/wtnetwork.py

```diff
@@ -51,7 +63,11 @@ def __init__(self, weights, thresholds=None, names=None, theta=None):
         :raises ValueError: if ``len(names)`` is not equal to the number of nodes
         :raises TypeError: if ``threshold_func`` is not callable
         """
-        self.weights = np.asarray(weights, dtype=np.float)
+        if type(weights) == int:
```

Comment:
I suggest using `isinstance(weights, int)` instead of comparing the type. Generally better form from what I understand as it handles inheritance better. Inheritance isn't really an issue here, but still...